testgen/src/validator.rs

```diff
@@ -45,19 +49,17 @@ impl Validator {
             bail!("validator identifier is too long")
         }
         bytes.extend(vec![0u8; 32 - bytes.len()].iter());
-        let seed = require_with!(
-            ed25519::Seed::from_bytes(bytes),
+        let secret = require_with!(
+            Ed25519SecretKey::from_bytes(&bytes).ok(),
             "failed to construct a seed from validator identifier"
         );
-        Ok(Ed25519Signer::from(&seed))
+        let public = Ed25519PublicKey::from(&secret);
+        Ok(Ed25519Keypair { secret, public })
     }
 
     /// Get a verifier from this validator companion.
-    pub fn get_verifier(&self) -> Result<Ed25519Verifier, SimpleError> {
-        let signer = self.get_signer()?;
-        let public_key = try_with!(signer.public_key(), "failed to get public key");
-        let verifier = Ed25519Verifier::from(&public_key);
-        Ok(verifier)
+    pub fn get_verifier(&self) -> Result<Ed25519PublicKey, SimpleError> {
```

Comment:
Likewise, perhaps rename this to:

```suggestion
    pub fn get_verifier(&self) -> Result<Ed25519Verifier, SimpleError> {
```

Reply:
:+1: